### PR TITLE
Add ability to set apiKey and applicationKey for AnalysisProviderDatadogConfig directly instead of reading files

### LINKED ISSUE #4492: Support setting apiKey and applicationKey for AnalysisProviderDatadogConfig directly instead of just via file
**What would you like to be added**:

Currently, the AnalysisProviderDatadogConfig configuration only supports auth information to be set via file (mount to the piped container). It would be great if we could support setting those values directly via fields like `apiKeyData` and `applicationKeyData` as for other configurations.

ref: https://pipecd.dev/docs-v0.41.x/user-guide/managing-piped/configuration-reference/#analysisproviderdatadogconfig

**Why is this needed**:

### PR DESCRIPTION
**Pull request by hungran**
**What this PR does / why we need it**:

add an option to set apiKey + applicationKey AnalysisProviderDatadogConfig directly in yaml file

**Which issue(s) this PR fixes**:

Fixes #4492

**Does this PR introduce a user-facing change?**: N/A

- **How are users affected by this change**: N/A
- **Is this breaking change**: N/A
- **How to migrate (if breaking change)**: N/A

### CODE REVIEW

**Comment by khanhtc1202** [comment]
@hungran Yes, it's. Thanks so much for contributions 😄

File: docs/content/en/docs-dev/user-guide/managing-piped/configuration-reference.md [hunk 1]
```diff
@@ -179,6 +179,8 @@ Must be one of the following structs:
 | address | string | The address of Datadog API server. Only "datadoghq.com", "us3.datadoghq.com", "datadoghq.eu", "ddog-gov.com" are available. Defaults to "datadoghq.com" | No |
 | apiKeyFile | string | The path to the api key file. | Yes |
 | applicationKeyFile | string | The path to the application key file. | Yes |
+| apiKey | string | Optional: API Key for Datadog API server. | No |
+| applicationKey | string | Optional: Application Key for Datadog API server | No |
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-piped/configuration-reference.md:183]:
```suggestion
| apiKeyData | string | Optional: API Key for Datadog API server. | No |
| applicationKeyData | string | Optional: Application Key for Datadog API server | No |
```
should add suffix `Data` to unify with other options like [this](https://github.com/pipe-cd/pipecd/blob/master/docs/content/en/docs-dev/user-guide/managing-piped/configuration-reference.md?plain=1#L218)

Comment (khanhtc1202) [reply docs/content/en/docs-dev/user-guide/managing-piped/configuration-reference.md:183]: Also as the piped convention, the field which holds key/token data should hold the base64 encode version of the key/token, so I think we should do the same here as well.

ref: https://pipecd.dev/docs-v0.44.x/user-guide/managing-piped/configuration-reference/#:~:text=Yes-,pipedKeyData,-string

**Comment by hungran** [comment]
> @hungran Thanks so much for contributions 😄
> 
> We should allow only one of those fields (if it provides the same data) to avoid misleading. Ref to other fields if you want reference 👍 https://github.com/pipe-cd/pipecd/blob/master/pkg/config/piped.go#L1101-L1115

Thanks for feedback, got your point, I added it

File: docs/content/en/docs-dev/user-guide/managing-piped/configuration-reference.md (hunk 1 above)
Comment (hungran) [reply docs/content/en/docs-dev/user-guide/managing-piped/configuration-reference.md]: yes, better if we hold key/token as base64 when direct on yaml file. Implemented that! 🙏

**Pull request merged**